### ISSUE TITLE
Clang++ asked me PLEASE submit this, so I did

### ISSUE THREAD

**Issue by csimon-bambecksystems** [llvm/llvm-project#60981]
Hey Clang++ project,

First, a meta-issue: The Clang++ output asks me to please attach the preprocessed source (.cpp) and run script (.sh) to the "bug report" or issue report if we are strict about GitHub terminology. However, GitHub does not allow me to attach these file types. My workaround is appending a .txt extension to each filename before uploading (because they are text anyway...).

Now, THE issue. Okay I am sorry, I do not know what is really going on here. But when my compiler tells very nicely to PLEASE, in all caps, submit a bug report to this GitHub page, you know guys, I just do what it asks me. Like I am being mugged or whatever. Please, I didn't mean to break the complier. I don't even know what I did. Just help me. The backtrace, preprocessed source, and run script are  politely attached. Sorry about the title. Anybody who knows something, just change it to a better one... thank you.

[output.txt](https://github.com/llvm/llvm-project/files/10827613/output.txt)
[laser_event_handlers-1410d9.sh.txt](https://github.com/llvm/llvm-project/files/10827646/laser_event_handlers-1410d9.sh.txt)
[laser_event_handlers-1410d9.cpp.txt](https://github.com/llvm/llvm-project/files/10827647/laser_event_handlers-1410d9.cpp.txt)

**Comment by DimitryAndric** [comment]
Yes, the inability to attach common source code file extensions in GitHub is a well-known shortcoming. Putting them in tarballs, zipfiles or indeed renaming them to .txt and then attaching is a fine workaround.

That said, I think you have found a shortcoming in the way clang generates the preprocessed sources: normally it attempts to make the .cpp file it writes on a crash completely self-contained, so that with the help of the shell script, somebody else on another machine can exactly reproduce the program.

However, in your case it looks like you are using C++ modules, and these are not included in the preprocessed source! I.e. when I attempt to run the command from the .sh file here locally:

```
$ clang -cc1 -triple x86_64-pc-windows-msvc19.35.32215 -emit-obj -mincremental-linker-compatible --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name laser_event_handlers.cpp -mrelocation-model pic -pic-level 2 -mframe-pointer=none -fmath-errno -ffp-contract=on -fno-rounding-math -ffp-exception-behavior=strict -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -fcoverage-compilation-dir=C:\\Users\\User\\Code\\SCA-7880\\master\\sim -O3 -std=c++20 -fdeprecated-macro -fdebug-compilation-dir=C:\\Users\\User\\Code\\SCA-7880\\master\\sim -ferror-limit 19 -fno-use-cxa-atexit -fms-extensions -fms-compatibility -fms-compatibility-version=19.35.32215 -fdelayed-template-parsing -fno-implicit-modules -fprebuilt-module-path=. -fcxx-exceptions -fexceptions -vectorize-loops -vectorize-slp -faddrsig -x c++ laser_event_handlers-1410d9.cpp
../src/laser_event_handlers.cpp:6:8: fatal error: module 'analyzer_id' not found
import analyzer_id;
~~~~~~~^~~~~~~~~~~
1 error generated.
```

Note that it does not crash, but also does not produce a working object file. If there is no useful way for you to provide the reference modules, I think you will have to attempt to reduce the test case to something smaller yourself.

**Comment by csimon-bambecksystems** [comment]
Thank you for the attention to my issue! Yes indeed I am using modules, so just to speculate, I feel like there is some weird interaction between my modules and the MSVC system headers as I began to see ODR errors where I though there should not be (or maybe I am just still confused how to use global module fragments to include system headers - though I was all straightened out about that, but maybe not?), and then the ODR errors mysteriously went away as I tweaked stuff, and then mysteriously came back, and finally, this crash began only after I split up one of my source module into partitions. I zipped up what I suppose to be the complete set of precompiled modules that the current test case depends on, LMK if you got what you need.

[precompiled-modules.zip](https://github.com/llvm/llvm-project/files/10829897/precompiled-modules.zip)

**Comment by csimon-bambecksystems** [comment]
I think this is a duplicate of #61063, and I closed that one today, so I'm going to close this one as well.
